---
title: "3.choosefor_methanotrophs"
author: "abbey yatsko"
date: "28/09/2022"
output: html_document
---

## The goal for this document is to sort out mound methane flux data and choose samples for sequencing 

Two main research questions: 
  1. How does the mound methanotroph community composition change across different termite species?
  2. For termite mounds with high inter-mound variation in methane flux, what is variation in inter-mound            methanotroph community composition?

### set workspace, read/clean data 
```{r}
library(ggplot2)
library(dplyr)
library(ggpubr)

mounds <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/finalfluxes/all_fluxes_mound.csv")

# select mounds from may22 campaign 
mounds <- mounds[mounds$campaign == "may22",]

# remove mounds that have fewer than 4 measurement positions per mound 
measure_number <- data.frame(table(mounds$sample))
measure_number <- subset(measure_number, Freq>3)
measure_number <- as.vector(measure_number$Var1)
mounds <- mounds[mounds$sample %in% measure_number,]
```

test if mound flux differs by species
```{r}
# one way ANOVA
aov.1 <- aov(flux.CH4 ~ species_s, data = mounds) # note experimental switch out of CH4 flux with CH4:CO2
summary(aov.1)
TukeyHSD(aov.1)

# assumptions
par(mfrow=c(2,2))
plot(aov.1)
par(mfrow=c(1,1))
```

###  To address question 1
Contrast 'big' and 'little' methane emitters
  * target top 4 and bottom 4 overall mound fluxes (pooled from all positions) for 3 species 
  * 8 mounds * 3 species * 4-5 positions 
```{r}
avgflux <- aggregate(mounds$flux.CH4, list(mounds$sample), FUN=mean)
y <- mounds[, c("sample",  "species_s")]

avgflux <- left_join(avgflux, y, by = c("Group.1"="sample"))
avgflux <- unique(avgflux)
avgflux$ species_s <- as.factor(avgflux$species_s)

# average methane flux by species
ggplot(data = avgflux, aes(x = species_s, y = x)) + 
  geom_boxplot()+ 
  ylab("CH4 flux umol/d/m2")+
  theme_light()

# pick out top 4 highest methane fluxes by species 
# (there is overlap between high and low of amitermes therefore shift from 5 to 4)
top <- avgflux %>%                                     
  arrange(desc(x)) %>% 
  group_by(species_s) %>%
  slice(1:4)

top$rank <- "top"

# pick out bottom 4 lowest methane fluxes by species
bottom <- avgflux %>%                                     
  arrange(x) %>% 
  group_by(species_s) %>%
  slice(1:4)

bottom$rank <- "bottom"

# remove entries without species info, clean up df
q1 <- rbind(top, bottom)
q1 <- q1 %>%
  rename(sample = Group.1, 
         avg_flux = x)

q1_all <- left_join(q1, mounds, by = c("sample", "species_s"))

# plot distribution of fluxes (not just average) 
ggplot(data = q1_all, aes(x = sample, y = flux.CH4, color =  as.factor(rank), fill = species_s)) + 
  geom_boxplot()+ 
  geom_jitter()+
  ylab("CH4 flux umol/d/m2")+
  theme_light()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  facet_grid(~ as.factor(species_s), 
             scales = "free_x",
             space = "free_x",
             switch = "x")+ 
  scale_fill_brewer(palette="YlOrRd")

# bar graph grouped by species - colored by top and bottom 
ggplot(q1, aes(x = reorder(as.factor(sample), -avg_flux), y = avg_flux, fill = rank)) + 
  geom_bar(stat='identity') +
  facet_grid(~ as.factor(species_s), 
             scales = "free_x",
             space = "free_x",
             switch = "x")+ 
  ylab("CH4 flux umol/d/m2")+
  xlab("")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

# plot for high values
q1H <- q1_all[q1_all$rank == c('top'),]
q1Hplot <- ggplot(data = q1H, aes(x = sample, y = flux.CH4, fill = species_s)) + 
  geom_boxplot()+ 
  geom_jitter()+
  ylab("CH4 flux umol/d/m2")+
  xlab("")+
  theme_classic()+
  theme( legend.position = "none")+
  facet_grid(~ as.factor(species_s), 
             scales = "free_x",
             space = "free_x",
             switch = "x")+ 
  scale_fill_brewer(palette="Blues")+
  ggtitle("Mounds with highest average CH4 flux")

# plot for low values 
q1L <- q1_all[q1_all$rank == c('bottom'),]
q1Lplot <- ggplot(data = q1L, aes(x = sample, y = flux.CH4, fill = species_s)) + 
  geom_boxplot()+ 
  geom_jitter()+
  ylab("CH4 flux umol/d/m2")+
  xlab("")+
  theme_classic()+
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none")+\
  theme(legend.position = "none")+
  facet_grid(~ as.factor(species_s), 
             scales = "free_x",
             space = "free_x",
             switch = "x")+ 
  scale_fill_brewer(palette="Blues")+ 
  ggtitle("Mounds with lowest average CH4 flux")

# overall sampling for Q1
ggarrange(q1Hplot, q1Lplot, ncol = 1)

# write.csv(q1, "/Users/abbeyyatsko/Downloads/q1.csv")
```

### To address question 2: 
explore intramound methane flux variation
target highest within-mound variation in methane flux and sequence each position individually 
  * top 5 most variable mounds * 3 species * 4-5 positions =  60-75 samples
```{r}
# calculate min and max flux for each sample 
data_min <- mounds %>%                                     
  group_by(sample) %>%
  summarise_at(vars(flux.CH4),
               list(min = min))
data_max <- mounds %>%                                     
  group_by(sample) %>%
  summarise_at(vars(flux.CH4),
               list(max = max))

minmax <- merge(data_max, data_min, by="sample")
minmax$range <- minmax$max - minmax$min 

# merge in species info 
minmax <- left_join(minmax, y, by = "sample")
minmax <- unique(minmax)

# range in intermound methane production by sample 
ggplot(data = minmax, aes(x = as.factor(species_s), y = range)) + 
  geom_boxplot()+ 
  ylab("Range CH4 flux umol/d/m2")+
  theme_light()

# choose 5 most variable mounds (greatest range)
q2 <- minmax %>%                                     
  arrange(desc(range)) %>% 
  group_by(species_s) %>%
  slice(1:5)

# boxplot of individual mounds
q2_samps <- as.vector(q2$sample)
q2_samps <- mounds[ mounds$sample %in% q2_samps, ]
length(unique(q2_samps$sample))

ggplot(data = q2_samps, aes(x = as.factor(sample), y = flux.CH4, fill = species_s)) + 
  geom_boxplot()+ 
  geom_jitter()+
  facet_grid(~ as.factor(species_s), 
             scales = "free_x",
             space = "free_x",
             switch = "x")+ 
  ylab("CH4 flux umol/d/m2")+
  xlab("")+
  theme_classic()+
  theme( legend.position = "none")+
  scale_fill_brewer(palette="Blues")

# see the overlap of samples in q1 and q2 
matches <- inner_join(q1, q2, 
                       c("sample" = "sample"))

# write.csv(q2, "/Users/abbeyyatsko/Downloads/q2.csv")
```

full sample lists 
```{r}
# create unique extract_ID that includes mound number, tag, and position of measurement
q1_samps <- q1_all
q1_samps$tag <- as.factor(q1_samps$tag)
q1_samps$extract_ID <- paste(q1_samps$sample,"-",q1_samps$tag, "-", q1_samps$position, sep = "")
q1_samps <- q1_samps %>% relocate(extract_ID, .after = sample)

# choose 3 samples per mound that are closest to the average 
# make column for difference from median in order to sort out easier 
q1_samps$avg_dif <- abs(q1_samps$avg_flux - q1_samps$flux.CH4)
q1_samps <- q1_samps %>% relocate(avg_dif, .after = avg_flux)

# choose the largest average difference grouped by sample 
q1_samps %>% group_by(sample) %>% top_n(1, avg_dif) -> a
a <- as.vector(a$extract_ID)
q1_samps <- q1_samps[!q1_samps$extract_ID %in% a,]

# subset out by species 
q1_samps_ami <- subset(q1_samps, species == "Ami")
q1_samps_copto <- subset(q1_samps, species == "Copto")
q1_samps_nasuti <- subset(q1_samps, species == "Nasuti")

# get rid of highest again for copto and nasuti (they have 5 measurements per mound usually, need to rid of an additional)
q1_samps_copto %>% group_by(sample) %>% top_n(1, avg_dif) -> copto_rm
copto_rm <- as.vector(copto_rm$extract_ID)
q1_samps_copto <- q1_samps_copto[!q1_samps_copto$extract_ID %in% copto_rm,]

q1_samps_nasuti %>% group_by(sample) %>% top_n(1, avg_dif) -> nasuti_rm
nasuti_rm <- as.vector(nasuti_rm$extract_ID)
q1_samps_nasuti <- q1_samps_nasuti[!q1_samps_nasuti$extract_ID %in% nasuti_rm,]

q1_samps_nasuti %>% 
  group_by(sample) %>% 
  slice_min(avg_dif, n = 3) -> q1_samps_nasuti

# bring species back together 
q1_samps <- rbind(q1_samps_ami, q1_samps_copto, q1_samps_nasuti)

# write.csv(q1_samps, "/Users/abbeyyatsko/Downloads/q1_samps.csv", row.names = FALSE)

# q1 minus overlaps: remove samples that are already being accounted for in q2 
# every position for mounds in q2 are being extracted
matches <- as.vector(matches$sample)
q1_samps_no_overlap <- q1_samps[!q1_samps$sample %in% matches,]
# write.csv(q1_samps_no_overlap, "/Users/abbeyyatsko/Downloads/q1_samps_no_overlap.csv", row.names = FALSE)

# q2: sequencing all individual samples within select mounds
dim(q2_samps)
# 68 samples (from 15 mounds)

# create unique extract_ID that includes mound number, tag, and position of measurement
q2_samps$tag <- as.factor(q2_samps$tag)
q2_samps$extract_ID <- paste(q2_samps$sample,"-",q2_samps$tag, "-", q2_samps$position, sep = "")
q2_samps <- q2_samps %>% relocate(extract_ID, .after = sample)

# export as .csv and check what overlap exists between q2 and q1
# write.csv(q2_samps, "/Users/abbeyyatsko/Downloads/q2_samps.csv", row.names = FALSE)

# total extractions: 115
```

summarize what each sample is for
```{r}
# q1: top/bottom emitters (select 3 samples per mound that are closest to median)
q1_samps$top_emitters <- 'y'
# write.csv(q1_samps, "/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/data_out/q1_top_emitters.csv", row.names = FALSE)

# q2: most variable mounds (select all samples)
q2_samps$variable_emitters <- 'y'

# write.csv(q2_samps, "/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/data_out/q2_variable_emitters.csv", row.names = FALSE)

# list of samples by question 
q1_out <- q1_samps %>% select(sample, extract_ID, species_s, top_emitters)
q2_out <- q2_samps %>% select(sample, extract_ID, species_s, variable_emitters)

# merge q1 and q2
x <- full_join(q1_out, q2_out, by = c("sample", "extract_ID", "species_s"))

# compare with sequencing metadata sheet
y <- read.csv("/Users/abbeyyatsko/Downloads/abb/sample_metadata.csv")

# join in Q1 metadata
q1_samples <- q1_samps %>% select(extract_ID, top_emitters, rank)
y <- left_join(y, q1_samples)
# MD41 is not here and it should be. need another bottom emitter for Amitermes - use MD3, tags 1, 2, 4

# MD3-1-mid, MD3-2-low, MD3-4-mid
y <- y %>%
  mutate(
    species_s = ifelse(extract_ID == "MD3-1-mid", "A. laurensis", species_s),
    top_emitters = ifelse(extract_ID == "MD3-1-mid", "y", top_emitters),
    rank = ifelse(extract_ID == "MD3-1-mid", "bottom", rank), 
    
    species_s = ifelse(extract_ID == "MD3-2-low", "A. laurensis", species_s),
    top_emitters = ifelse(extract_ID == "MD3-2-low", "y", top_emitters),
    rank = ifelse(extract_ID == "MD3-2-low", "bottom", rank),
    
    species_s = ifelse(extract_ID == "MD3-4-mid", "A. laurensis", species_s),
    top_emitters = ifelse(extract_ID == "MD3-4-mid", "y", top_emitters),
    rank = ifelse(extract_ID == "MD3-4-mid", "bottom", rank)
  )

# join in Q2 metadata
q2_samples <- q2_samps %>% select(extract_ID, species_s, variable_emitters)
y <- left_join(y, q2_samples, by = c("extract_ID"))
y <- y %>%
  mutate(species = coalesce(species_s.x, species_s.y)) %>%
  select(-species_s.x, -species_s.y)

# MD39 only has 4 measurement points even though it is a N. magnus (which usually has 5)
# MD51 only has 3 measurement points even though it is a N. magnus (which usually has 5)

# MD39-3-mid should be included in variable emitters
# MD50 add to top_emitters (MD50-2-low, MD50-3-low, MD50-4-mid)
# MD38 add as variable_emitters (even though it is not really a top variable emitter)
y <- y %>%
  mutate(
    species = ifelse(extract_ID == "MD39-3-mid", "N. magnus", species),
    variable_emitters = ifelse(extract_ID == "MD39-3-mid", "y", variable_emitters), 
    
    species = ifelse(extract_ID == "MD50-2-low", "N. magnus", species),
    top_emitters = ifelse(extract_ID == "MD50-2-low", "y", top_emitters),
    rank = ifelse(extract_ID == "MD50-2-low", "top", rank), 
    
    species = ifelse(extract_ID == "MD50-3-low", "N. magnus", species),
    top_emitters = ifelse(extract_ID == "MD50-3-low", "y", top_emitters),
    rank = ifelse(extract_ID == "MD50-3-low", "top", rank), 
    
    species = ifelse(extract_ID == "MD50-4-mid", "N. magnus", species),
    top_emitters = ifelse(extract_ID == "MD50-4-mid", "y", top_emitters),
    rank = ifelse(extract_ID == "MD50-4-mid", "top", rank), 
    
    species = ifelse(sample == "MD38", "N. magnus", species),
    variable_emitters = ifelse(sample == "MD38", "y", variable_emitters)
  )

# add in variable_emitters range values
u <- minmax %>% arrange(desc(range)) %>% 
    group_by(species_s) %>%
    mutate(range_rank = row_number()) %>%
    select(range, range_rank, sample)

y <- left_join(y, u, by = c("sample" = "sample"))

# export metadata with relevent corresponding question information
# write.csv(y, "/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/data_out/metagenomes_metadata.csv", row.names = FALSE)
```

```{r}

variable <- y %>% filter(variable_emitters == 'y')
top <- y %>% filter(top_emitters == 'y')
```


