---
title: "\"
author: "abbey yatsko"
date: "2025-06-13"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

libraries 
```{r}
library(ggplot2)
library(emmeans)
library(car)
library(dplyr)
library(vegan)
library(lmerTest)
```

data 
```{r}
individuals <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/finalfluxes/p_fluxes_f.csv")
all_mound_avg <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/finalfluxes/all_fluxes_mound_avg.csv")

resampled_mound_avg <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/finalfluxes/all_fluxes_resample_reps_avg.csv")

bac <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/metagenomes/Bact_tax_abun.csv")
sample_metadata <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/metagenomes/sample_metadata.csv")
seq_metadata <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/metagenomes/metadata-15165487-processed-ok.csv")
fluxes <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/finalfluxes/all_fluxes_mound.csv")
methanotroph <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/metagenomes/methanotroph_genera.csv")
flux_relabund_pmoa <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/metagenomes/flux_relabund.csv")
highlow_emit <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/data_out/q1_top_emitters.csv")
var_emit <- read.csv("/Users/abbeyyatsko/Desktop/repos/TMI_flux/data/data_out/q2_variable_emitters.csv")
```

Result 1. Comparison of individual termite emissions and average mound emissions 

clean data
```{r}
individuals$species <- as.factor(indiividuals$species)

all_mound_avg$ID_mound <- as.factor(all_mound_avg$ID_mound)
all_mound_avg$species_s <- as.factor(all_mound_avg$species_s)
all_mound_avg <- all_mound_avg[all_mound_avg$mean_mound_fluxCH4 > 0, ]
```

analysis 
```{r}
# individuals
mod <- aov(log(TEF_ug_g_h) ~ species, data = individuals)
summary(mod) 
emmeans(mod, pairwise ~ species, adjust = "tukey") 

# average individual emissions
individuals %>%
  group_by(species) %>%
  summarise(mean_mound_fluxCH4 = mean(TEF_ug_g_h, na.rm = TRUE)) %>%
  ungroup()

# mounds
mod <- lm(log(mean_mound_fluxCH4*0.6683) ~ species_s, data = all_mound_avg)
Anova(mod) 
emmeans(mod, pairwise ~ species_s, adjust = "tukey") 

all_mound_avg %>%
  group_by(species_s) %>%
  # mound flux is umol/mound/d convert to Tg/mound/yr
  summarise(whole_mound_fluxCH4 = mean(mean_mound_fluxCH4*0.6683, na.rm = TRUE))

# landscape scale (for resampled mounds to capture season variation)
landscape_flux <- resampled_mound_avg %>%
  group_by(species_s) %>%
  # mound flux is umol/mound/d convert to Tg/mound/yr
  summarise(whole_mound_fluxCH4 = mean(mound_CH4flux*5.857e-15, na.rm = TRUE), 
            whole_mound_fluxCH4_sd = sd(mound_CH4flux*5.857e-15, na.rm = TRUE)) %>%
  mutate(
    density = case_when(
      species_s == "A. laurensis" ~ 120.4,
      species_s == "C. acinaciformis" ~ 2.8,
      species_s == "N. magnus" ~ 5.6,
      TRUE ~ NA_real_
    ),
    flux_density_scaled = whole_mound_fluxCH4 * density,
    flux_density_sd_scaled = whole_mound_fluxCH4_sd * density
  ) %>%
  ungroup()

```

Figure 1. Comparison of individual termite emissions and average mound emissions 
```{r}
a <- ggplot(individuals, aes(x = species, y = TEF_ug_g_h, fill = species))+
  geom_boxplot(outlier.shape = NA)+
  ylab("Termite EF (ug CH4/hr/g termite)")+
  xlab("")+
  scale_x_discrete(labels = c("A. laurensis", "C. acinaciformis", "N. magnus"))+
  scale_fill_viridis_d(option = "D") +
  theme_classic()+
  ggtitle("a)") + 
  theme(legend.position = "none")
b <- ggplot(all_mound_avg, aes(x = species_s, y = mean_mound_fluxCH4*0.6683, fill = species_s)) + 
  geom_boxplot()+ 
  scale_x_discrete(labels = c("A. laurensis", "C. acinaciformis", "N. magnus"))+
  theme_classic()+
  ylab("Mean mound CH4 flux (ug CH4/hr/m2)")+
  xlab("")+
  ggtitle("b)") + 
  scale_fill_viridis_d(option = "D") +
  theme(legend.position = "none")
c <- ggplot(landscape_flux, aes(x = species_s, y = flux_density_scaled, fill = species_s)) +
  geom_bar(stat = "identity", width = 0.6) +
  geom_errorbar(
    aes(ymin = flux_density_scaled - flux_density_sd_scaled,
        ymax = flux_density_scaled + flux_density_sd_scaled),
    width = 0.2
  ) +
  scale_fill_viridis_d(option = "D") +
  labs(
    x = "Species",
    y = expression("Density-scaled CH"[4]*" flux (Tg / yr)"),
    title = "Total CH4 Flux per Species (scaled by mound density)"
  ) +
  theme_classic() +
  theme(legend.position = "none")
ggarrange(a,b,c, ncol = 3)
```

Result 2. SEM build

clean data 
```{r}
# duplicate wall thickness entry for each unique sample
resampled_mound_avg <- resampled_mound_avg %>%
  group_by(sample) %>%
  mutate(wall_thickness_mm = unique(na.omit(wall_thickness_mm))[1]) %>%
  ungroup()

# centered and scaled mound-level variables 
resampled_mound_avg$wt_mm_st <- ave(resampled_mound_avg$wall_thickness_mm, resampled_mound_avg$species_s, FUN = scale)
resampled_mound_avg$volume_m3_st <- ave(resampled_mound_avg$volume_m3_final, resampled_mound_avg$species_s, FUN = scale)

# variable structure
resampled_mound_avg$species_s <- as.factor(resampled_mound_avg$species_s)
resampled_mound_avg$sample <- as.factor(resampled_mound_avg$sample)
resampled_mound_avg$campaign <- as.factor(resampled_mound_avg$campaign)
```

analysis
```{r}
vars_needed <- c("mean_mound_fluxCH4", "species_s", "volume_m3_st",
                 "wt_mm_st", "temp", "campaign", "sample")
clean_data <- na.omit(resampled_mound_avg[ , vars_needed])

mod1 <- lm(volume_m3_st ~ species_s, data = clean_data)
mod2 <- lm(wt_mm_st ~ species_s, data = clean_data)
mod3 <- lmer(mean_mound_fluxCH4 ~ volume_m3_st + wt_mm_st + species_s +
               temp + campaign + (1 | sample), data = clean_data)

sem_model <- psem(
  mod1,
  mod2,
  mod3
)

summary(sem_model) 
```

Result 3. Influence of methanotroph communities and genes 

clean data
```{r}
# subset sequencing and sample metadata, then join
seq_metadata_x <- seq_metadata[, c("accession", "sample_name")]
sample_metadata_x <- sample_metadata[, c("sequencing_id", "sample_id", "extract_ID")]
colnames(sample_metadata_x)[2] <- "sample_name"
ids <- left_join(
  seq_metadata_x,
  sample_metadata_x,
  by = "sample_name"
)

# join with flux data 
fluxes <- fluxes[fluxes$campaign == "may22", ]
fluxes$extract_ID <- paste(fluxes$sample,"-",fluxes$tag, "-", fluxes$position, sep = "")
x <- fluxes[, c("ID_measurement", "extract_ID", "flux.CH4", "avg_respT", "species_s")]

microbes_fluxes <- left_join(ids, x, by = "extract_ID")

# remove MD8 as it's all negative (error in my selection)
microbes_fluxes <- microbes_fluxes[!grepl("MD8", microbes_fluxes$ID_measurement) , ]

# remove rows where flux.CH4 is NA
microbes_fluxes <- microbes_fluxes[!is.na(microbes_fluxes$flux.CH4), ]

# pivot bacteria df longer so that all SRR are rows 
bac_long <- bac %>%
  pivot_longer(
    cols = starts_with("SRR"),
    names_to = "accession",
    values_to = "abundance"
  )

microbes_fluxes_bac <- left_join(microbes_fluxes, bac_long,  by = "accession") 

# sum of reads based on sample_name
total_reads <- microbes_fluxes_bac %>%
  group_by(sample_name, extract_ID, ID_measurement, flux.CH4, avg_respT, species_s) %>%
  summarise(total_reads = sum(abundance), .groups = 'drop')

# filter out for methanotrophic genera
methanotroph$genus <- paste0("g__", methanotroph$genus)
out_m <- microbes_fluxes_bac %>%
  filter(Genus %in% methanotroph$genus) %>%
  filter(!is.na(flux.CH4))

out_m$mound_ID <- sub("-.*", "", out_m$extract_ID)
```

define datasets for high/low emitters 
```{r}
highlow <- highlow_emit["extract_ID"]
var <- var_emit["extract_ID"]

out_m_hl <- out_m %>%
  filter(extract_ID %in% highlow$extract_ID)

out_m_var <- out_m %>%
  filter(extract_ID %in% var$extract_ID)

# number of unique mounds (with enough data) for each question 
out_m_var %>%
     group_by(species_s) %>%
     summarise(n_mounds = n_distinct(mound_ID)) %>%
     arrange(desc(n_mounds))
```

prep data for NMDS 
```{r}
# create X and Y matrix of data (all samples)
out_m_hl %>% dplyr::select(accession, Genus, abundance) %>%
  filter(!(accession %in% c("SRR32638406")) & !(Genus == "g__")) %>% # outlier based on original nmds plot
  group_by(accession, Genus) %>%
  summarise(abundance = sum(abundance)) %>%
  ungroup() %>%
  mutate(abundance = as.numeric(abundance)) %>%
  pivot_wider(names_from = Genus, values_from = abundance, id_cols = accession) %>%
  column_to_rownames("accession") -> Y_matrix
out_m_hl %>% dplyr::select(accession, Genus, species_s, abundance, flux.CH4, mound_ID) %>%
  filter(!(accession %in% c("SRR32638406")) & !(Genus == "g__")) %>% # outlier based on original nmds plot
  group_by(accession, Genus, species_s, flux.CH4, mound_ID) %>%
  summarise(abundance = sum(abundance)) %>%
  pivot_wider(names_from = Genus, values_from = abundance)-> X_matrix

# Transform data
spe.h <- decostand(Y_matrix, "hellinger")

# Calculate distance matrix
spe_distmat <- vegdist(spe.h, method = "bray") %>%  as.matrix()
```

NMDS analysis 
```{r}
# Running NMDS metaMDS
nmds_output <- metaMDS(spe_distmat, distance = "bray", k = 2, maxit = 999, trymax = 500, wascores = T)

# Rownames
accession <- rownames(Y_matrix) 

# Extracting NMDS scores
data.scores <- as.data.frame(scores(nmds_output, display="sites"))  %>% 
  as_tibble() %>% 
  add_column(accession)

# Adding metadata to NMDS scores
nmds_scores_df <- left_join(data.scores, X_matrix, by = "accession") 
nmds_scores_df <- left_join(nmds_scores_df, seq_metadata_x, by = "accession")
nmds_scores_df <- left_join(nmds_scores_df, sample_metadata_x, by = "sample_name")
nmds_scores_df <- left_join(nmds_scores_df, flux_relabund_pmoa)
nmds_scores_df$variable_emitters <- as.factor(ifelse(is.na(nmds_scores_df$variable_emitters), "n", nmds_scores_df$variable_emitters))

# sum of all methanotrophs per sample
nmds_scores_df <- nmds_scores_df %>%
  mutate(meth_sum = rowSums(across(starts_with("g__")))) %>%
  mutate(shannon = diversity(dplyr::select(., starts_with("g__")), index = "shannon"))

# remove if there are NAs in pmoa
nmds_scores_df <- nmds_scores_df %>%
  filter(!is.na(resampled_relabund))

# Extract species scores
spp.fit <- envfit(nmds_output, Y_matrix)

# ggplot plot - species scores
spp.scrs <- as.data.frame(scores(spp.fit, display = "vectors")) #save species intrinsic values into dataframe
spp.scrs <- cbind(spp.scrs, Genus = rownames(spp.scrs)) 
spp.scrs <- cbind(spp.scrs, pval = spp.fit$vectors$pvals) #add pvalues to dataframe so you can select species which are significant

#spp.scrs<- cbind(spp.scrs, abrev = abbreviate(spp.scrs$Species, minlength = 6)) #abbreviate species names
sig.spp.scrs <- subset(spp.scrs, pval<=0.05) #subset data to show species significant at 0.05
sig.spp.scrs$Genus <- sub("^g__", "", sig.spp.scrs$Genus)
arrow_scaling <- 0.5
sig.spp.scrs$NMDS1s <- sig.spp.scrs$NMDS1 * arrow_scaling
sig.spp.scrs$NMDS2s <- sig.spp.scrs$NMDS2 * arrow_scaling
nmds_scores_df$NMDS1.2 <- sqrt(nmds_scores_df$NMDS1^2 + nmds_scores_df$NMDS2^2)

# join with total_reads
nmds_scores_df <- left_join(nmds_scores_df, total_reads)

# calculate relative abundance for all columns beginning with g__ and total methanotroph count
nmds_scores_df <- nmds_scores_df %>%
  mutate(across(starts_with("g__"), ~ . / total_reads, .names = "relabund_{col}")) %>%
    mutate(relabund_meth_sum = (across(starts_with("meth_sum"))[[1]]) / total_reads)

# match y_matrix and nmds_scores_df 
Y_matrix_matched <- Y_matrix[rownames(Y_matrix) %in% nmds_scores_df$accession, ]
Y_matrix_matched <- Y_matrix[nmds_scores_df$accession, ]

X_matrix_matched <- X_matrix[X_matrix$accession %in% nmds_scores_df$accession, ]

# PERMANOVA: does community composition change for species or flux
mod_adon <- adonis2(Y_matrix_matched ~ species_s + flux.CH4, permutations = 999, 
                    data = X_matrix_matched, 
                    by = "terms")

# Multivariate Dispersion Test: 
mod <- betadisper(vegdist(decostand(Y_matrix_matched, 'hellinger')),
                  interaction(nmds_scores_df$mound_ID, nmds_scores_df$variable_emitters), 
                  bias.adjust=T)
anova(mod)

# testing interaction between NMDS axes and relabund PMOA
mod_int <- lmer(resampled_relabund_scaled ~ NMDS1 + NMDS2 + (1|mound_ID), data = nmds_scores_df)
summary(mod_int)

# effect of pmoA abundance on CH4 emission 
nmds_scores_df$mound_ID <- as.factor(nmds_scores_df$mound_ID)
mod <- lmer(flux.CH4 ~ resampled_relabund_scaled + species_s + NMDS1 * NMDS2 + relabund_meth_sum +  (1|mound_ID), data = nmds_scores_df)
summary(mod)

# by individual genus (relative abundance)
# list of unique things begining with "relabund"
relabund_cols <- grep("^relabund_g", names(nmds_scores_df), value = TRUE)

# loop through each relative abundance column and run a model
for (x in relabund_cols) {
  cat("Model for", x, ":\n")
  mod <- lmer(flux.CH4 ~ get(x) + species_s + (1|mound_ID), data = nmds_scores_df)
  print(summary(mod))
}

ggplot(nmds_scores_df, aes(x = relabund_g__Methylocystis, y = flux.CH4)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  labs(x = "Relative Abundance of Methannotroph genera", y = "CH4 Flux") +
  theme_minimal()

# for variable mounds 
# test community composition, pmoa, total methanotroph abundance, species on flux (response) 
mod <- lmer(flux.CH4 ~ resampled_relabund_scaled + species_s + NMDS1 * NMDS2 + relabund_meth_sum + (1|mound_ID), data = nmds_scores_df)
summary(mod)

# group by mound_ID and calculate the range for each column starting with g__
nmds_scores_df <- nmds_scores_df %>%
  group_by(mound_ID) %>%
  mutate(across(starts_with("g__"), ~ max(.) - min(.), .names = "range_{col}")) %>%
  mutate(relabund_meth_sum = meth_sum/total_reads) %>%
  mutate(across(starts_with("relabund_meth_sum"), ~ max(.) - min(.), .names = "range_{col}")) %>%
  mutate(std_range_CH4 = range/mean(flux.CH4)) %>%
  mutate(std_range_meth_sum = range_relabund_meth_sum/mean(relabund_meth_sum)) %>%
  ungroup()

ggplot(nmds_scores_df, aes(x = std_range_meth_sum, y = std_range_CH4)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  labs(x = "Range of Abundance of Summed Methannotroph genera", y = "standardised CH4 Flux range") +
  theme_minimal()

mod <- lmer(std_range_CH4 ~ std_range_meth_sum + species_s + (1|mound_ID), data = nmds_scores_df)
summary(mod)
```

Figure 3. Community composition and gene influence on CH4 flux
```{r}
# NMDS plot 
a <- ggplot(data = nmds_scores_df) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = species_s), size = 4) +
  stat_ellipse(aes(x = NMDS1, y = NMDS2, color = species_s), alpha = 0.6, level = 0.95) +
  geom_segment(
    data = sig.spp.scrs, 
    aes(x = 0, xend = NMDS1s*.4, y = 0, yend = NMDS2s*.4), 
    arrow = arrow(length = unit(0.25, "cm")), 
    colour = "grey10", lwd = 0.3
  ) +
  ggrepel::geom_text_repel(
    data = sig.spp.scrs, 
    aes(x = NMDS1s*.4, y = NMDS2s*.4, label = Genus), 
    cex = 3, direction = "both", segment.size = 0.25
  ) +
  scale_color_viridis_d(option = "D") +  # apply viridis discrete palette
  theme_classic()

# interaction of NMDS1 & NMDS2 on pmoA 
mod_int <- lmer(resampled_relabund_scaled ~ NMDS1 * NMDS2 + (1|mound_ID), data = nmds_scores_df)
b_pred <- ggpredict(mod_int, terms = c("NMDS1"))# %>% plot()

sm <- "#fce725"
med <- '#20908c'
lg <- '#440153'
v <- c(-0.02, 0, 0.02)

b <- ggplot(nmds_scores_df, aes(x = NMDS1, y = resampled_relabund_scaled, color = NMDS2)) +
  geom_jitter() +

  # -0.02 group
  geom_ribbon(data = filter(b_pred, group == '-0.02'),
              aes(x = x, ymin = conf.low, ymax = conf.high),
              inherit.aes = FALSE, fill = sm, alpha = 0.2) +
  geom_line(data = filter(b_pred, group == '-0.02'),
            aes(x = x, y = predicted),
            inherit.aes = FALSE, colour = sm) +

  # 0 group
  geom_ribbon(data = filter(b_pred, group == '0'),
              aes(x = x, ymin = conf.low, ymax = conf.high),
              inherit.aes = FALSE, fill = med, alpha = 0.2) +
  geom_line(data = filter(b_pred, group == '0'),
            aes(x = x, y = predicted),
            inherit.aes = FALSE, colour = med) +

  # 0.02 group
  geom_ribbon(data = filter(b_pred, group == '0.02'),
              aes(x = x, ymin = conf.low, ymax = conf.high),
              inherit.aes = FALSE, fill = lg, alpha = 0.2) +
  geom_line(data = filter(b_pred, group == '0.02'),
            aes(x = x, y = predicted),
            inherit.aes = FALSE, colour = lg) +

  scale_color_viridis_c(option = "D") +
  theme_classic()


ggarrange(a, b)

# plot of methanotrophs relative abundance 
relabund_cols <- nmds_scores_df[,grep("^relabund_g__", names(nmds_scores_df), value = TRUE)]
relabund_long <- relabund_cols %>%
  pivot_longer(
    cols = everything(),
    names_to = "methanotroph",
    values_to = "relative_abundance")

ggplot(relabund_long, aes(x = methanotroph, y = relative_abundance, fill = methanotroph)) +
  geom_boxplot() +
  labs(x = "Methanotroph genera", y = "Relative abundance", fill = "Methanotroph genera") +
  theme_classic() +
  theme(legend.position = "none") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_fill_viridis_d(option = "D")


```

